docs/resources/application_key.md

```diff
@@ -13,9 +13,23 @@ Provides a Datadog Application Key resource. This can be used to create and mana
 ## Example Usage
 
 ```terraform
-# Create a new Datadog Application Key
-resource "datadog_application_key" "foo" {
-  name = "foo-application"
+# Source the permissions for scoped keys
+data "datadog_permissions" "dd_perms" {}
+
+# Create an unrestricted Application Key
+# This key inherits all permissions of the user that owns the key
+resource "datadog_application_key" "unrestricted_key" {
+  name = "Unrestricted Application Key"
+  # scopes unset - inherits all user permissions
+}
+
+# Create a scoped Application Key for monitor management
+resource "datadog_application_key" "monitor_management_key" {
+  name = "Monitor Management Key"
+  scopes = [
+    data.datadog_permissions.dd_perms.permissions.monitors_read,
+    data.datadog_permissions.dd_perms.permissions.monitors_write
+  ]
```

Comment:
FYI @anthony-dgx I find the current TF example for application keys limiting, would like to flesh it out a bit